### PR TITLE
Adapt "gitdocs serve" to work in a remote dev env

### PR DESCRIPTION
**Pull request by ledestin**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

At the moment, `gitdocs serve` uses `0.0.0.0` as `host`, and it's broken if accessed from a remote box. I made `gitdocs` use `APP_DOMAIN` as `host` dynamically.

## Related Tickets & Documents

#7706

## QA Instructions, Screenshots, Recordings

Have `APP_DOMAIN` set in `application.yml` to a domain you want `gitdocs` to listen on. If `APP_DOMAIN` not defined, that's `localhost`.

## Added tests?

- [ ] yes
- [x] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

### CODE REVIEW

File: .gitdocs.js [hunk 1]
```diff
@@ -7,5 +7,20 @@ var config = require('./docs/.gitdocs.json');
...
 module.exports = new Promise((resolve, reject) => {
   config.root = 'docs/';
+  setupHost(config);
   resolve(config);
 });
+
+function fetchAppDomain() {
+  const { execSync } = require('child_process');
+
+  return execSync('rake app_domain').toString().replace(/\s/g, '');
```
Comment (citizen428) [inline .gitdocs.js:17]: I don't think we need a rake tasks for this, we can just use the Rails runner:

```sh
$ rails runner 'puts ApplicationConfig.app_domain_no_port'
localhost
```
```suggestion
  return execSync('rails runner 'puts ApplicationConfig.app_domain_no_port'').toString().replace(/\s/g, '');
```

Comment (ledestin) [reply .gitdocs.js]: Thanks for the suggestion and providing the fix! Always a pleasure working with you.

**Comment by ledestin** [comment]
Updated, and removed the rake task.

File: .gitdocs.js [hunk 2]
```diff
@@ -7,5 +7,20 @@ var config = require('./docs/.gitdocs.json');
...
 module.exports = new Promise((resolve, reject) => {
   config.root = 'docs/';
+  setupHost(config);
   resolve(config);
 });
+
+function fetchAppDomain() {
+  const { execSync } = require('child_process');
+
+  return execSync('rails runner 'puts ApplicationConfig.app_domain_no_port'').toString().replace(/\s/g, '');
```
Comment (maestromac) [inline .gitdocs.js:17]: Perhaps you meant to put a single quote within the double quote here? ie `"rails runner 'put...' "`

Once I made that change locally, I can't run `gitdoc serve". The following is my error
```
events.js:288
      throw er; // Unhandled 'error' event
      ^

Error: getaddrinfo ENOTFOUND DEPRECATIONWARNING:Defaultvalueswillberemovedinthenextminor-releaseofENVied(i.e.>v0.9).Formoreinfoseehttps://gitlab.com/envied/envied/tree/0-9-releases#defaults.(calledfromblockinloadat/Users/macsiri/dev.to/Envfile:4)DEPRECATIONWARNING:InitializationautoloadedtheconstantsAudit,Audit::Helper,Audit::Subscribe,Aws::FakeClient,RateLimitChecker,andCacheBuster.Beingabletodothisisdeprecated.AutoloadingduringinitializationisgoingtobeanerrorconditioninfutureversionsofRails.Reloadingdoesnotreboottheapplication,andthereforecodeexecutedduringinitializationdoesnotrunagain.So,ifyoureloadAudit,forexample,theexpectedchangeswon'tbereflectedinthatstaleModuleobject.`config.autoloader`issetto`classic`.Theseautoloadedconstantswouldhavebeenunloadedif`config.autoloader`hadbeensetto`:zeitwerk`.Please,checkthe"AutoloadingandReloadingConstants"guideforsolutions.(calledfrom<main>at/Users/macsiri/dev.to/config/environment.rb:5)**[Honeybadger]InitializingHoneybadgerErrorTrackerforRuby.Shipit!version=4.7.0framework=railslevel=1pid=30910**[Honeybadger]Developmentmodeisenabled.Datawillnotbereporteduntilyoudeployyourapp.level=2pid=30910localhost
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:64:26)
Emitted 'error' event on WebSocketServer instance at:
    at Server.emit (events.js:311:20)
    at GetAddrInfoReqWrap.doListen [as callback] (net.js:1493:12)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:64:17) {
  errno: 'ENOTFOUND',
  code: 'ENOTFOUND',
  syscall: 'getaddrinfo',
  hostname: 'DEPRECATIONWARNING:Defaultvalueswillberemovedinthenextminor-releaseofENVied(i.e.>v0.9).Formoreinfoseehttps://gitlab.com/envied/envied/tree/0-9-releases#defaults.(calledfromblockinloadat/Users/macsiri/dev.to/Envfile:4)DEPRECATIONWARNING:InitializationautoloadedtheconstantsAudit,Audit::Helper,Audit::Subscribe,Aws::FakeClient,RateLimitChecker,andCacheBuster.Beingabletodothisisdeprecated.AutoloadingduringinitializationisgoingtobeanerrorconditioninfutureversionsofRails.Reloadingdoesnotreboottheapplication,andthereforecodeexecutedduringinitializationdoesnotrunagain.So,ifyoureloadAudit,forexample,theexpectedchangeswon\'tbereflectedinthatstaleModuleobject.`config.autoloader`issetto`classic`.Theseautoloadedconstantswouldhavebeenunloadedif`config.autoloader`hadbeensetto`:zeitwerk`.Please,checkthe"AutoloadingandReloadingConstants"guideforsolutions.(calledfrom<main>at/Users/macsiri/dev.to/config/environment.rb:5)**[Honeybadger]InitializingHoneybadgerErrorTrackerforRuby.Shipit!version=4.7.0framework=railslevel=1pid=30910**[Honeybadger]Developmentmodeisenabled.Datawillnotbereporteduntilyoudeployyourapp.level=2pid=30910localhost'
}
```

Comment (ledestin) [reply .gitdocs.js]: @maestromac thanks, I should've checked. Fixed now.

**Pull request merged**